src/main/java/androidx/util/Base64.kt

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2018 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@file:Suppress("NOTHING_TO_INLINE")
+
+package androidx.util
+
+import android.util.Base64
+import java.nio.charset.Charset
+
+/**
+ * Convert [String] to base-64 encoded [String].
+ */
+inline fun String.toBase64(
+    offset: Int = 0,
+    len: Int = this.length,
```

Comment:
Length won't work here as you are using it as the length of the bytes whereas the parameter represents the number of characters. You need to do a `substring` before calling `toByteArray` and then pass the length of the byte array to `Base64`.

I would also write a test for this with two multi-byte UTF-8 characters in the source string and then a length of 1 which should thus encode more than one byte.

Reply:
I updated this PR with everything else.

How would this substring work?

Reply:
Any update on how you want this to work? Or do you mind if I just make the PR for the String to Base64 and Base64 to String?